### ISSUE TITLE
try to make reproducible examples (missing cargo dependencies)

### ISSUE THREAD

**Issue by nbari** [rust-lang/async-book#91]
As a newbie to Rust, I found very complex to follow documents in where part of the code belongs to the `std` library and other depends on `crates` from example from: [04_async_await_primer](https://rust-lang.github.io/async-book/01_getting_started/04_async_await_primer.html)

> async/.await is Rust's built-in tool for writing asynchronous functions that look like synchronous code. async transforms a block of code into a state machine that implements a trait called Future. Whereas calling a blocking function in a synchronous method would block the whole thread, blocked Futures will yield control of the thread, allowing other Futures to run.
> To create an asynchronous function, you can use the async fn syntax:
> `async fn do_something() { /* ... */ }`
> The value returned by async fn is a Future. For anything to happen, the Future needs to be run on an executor.

Then there is this example:

```rust
// `block_on` blocks the current thread until the provided future has run to
// completion. Other executors provide more complex behavior, like scheduling
// multiple futures onto the same thread.
use futures::executor::block_on;

async fn hello_world() {
    println!("hello, world!");
}

fn main() {
    let future = hello_world(); // Nothing is printed
    block_on(future); // `future` is run and "hello, world!" is printed
}
```

By trying to **learn by doing**,  I do:

    cargo new example --bin

Then go and edit `src/main.rs` copy, paste code from the examples and try to run it with:

    cargo run

In this case I just stumble with an error regarding `undeclare type of module futures`:

```
error[E0433]: failed to resolve: use of undeclared type or module `futures`
 --> src/main.rs:4:5
  |
4 | use futures::executor::block_on;
  |     ^^^^^^^ use of undeclared type or module `futures`
```

Initial toughs are to search about `futures` and by guessing I try  to add in the dependencies of `Cargo.toml` something like this:

```
[dependencies]
futures = "0.3"
```

Code now compiles, I can run it but I am more confused since I found out that there is also another `future` from the `std` library: https://doc.rust-lang.org/std/future/index.html

It will help if the examples could be improved by explaining what dependencies are used and why the usage of `crates` instead of the `std` libraries.

In [03_state_of_async_rust.html](https://rust-lang.github.io/async-book/01_getting_started/03_state_of_async_rust.html) there is a hint about the crate, but maybe in the comments of the code, it could help to have  something basic like:

```
/// check https://rust-lang.github.io/async-book/01_getting_started/03_state_of_async_rust.html
/// Depends on `futures = "0.3"` (check Cargo.toml [dependencies])
...
```

**Comment by taiki-e** [comment]
I think #89 fixed this.

**Comment by taiki-e** [comment]
Closing -- fixed by #89 and #93
